mldsa/sign.c

```diff
@@ -155,19 +155,140 @@ __contract__(
   shake256_squeeze(out, outlen, &state);
 }
 
+#define NONCE_UB ((UINT16_MAX - MLDSA_L) / MLDSA_L)
+
+/*************************************************
+ * Name:        attempt_signature_generation
+ *
+ * Description: Attempts to generate a single signature.
+ *              This function encapsulates one iteration of the
+ *              rejection sampling process to help CBMC analysis.
+ *
+ * Arguments:   - uint8_t *sig: pointer to output signature
+ *              - const uint8_t *mu: pointer to message hash
```

Comment:
`message hash` is a little imprecise here and may confuse reader. Mu is not just a message hash, but a prepared premessage string that's hashed. But also, this isn't always `Mu`, this function gets called from sign internal, and that can sign either `Mu` or raw message types.